core/src/main/java/software/amazon/awssdk/core/protocol/json/SdkStructuredCborFactory.java

```diff
@@ -33,15 +34,18 @@
  * Creates generators and protocol handlers for CBOR wire format.
  */
 @SdkInternalApi
-final class SdkStructuredCborFactory {
+public abstract class SdkStructuredCborFactory {
 
-    private static final JsonFactory CBOR_FACTORY = new CBORFactory();
+    protected static final JsonFactory CBOR_FACTORY = new CBORFactory();
+
+    protected static final BiFunction<JsonFactory, String, StructuredJsonGenerator> GENERATOR_SUPPLIER =
```

Comment:
Might be advantageous to have a subinterface of BiFunction that gives the params better names. Didn't immediately grok what String could be without looking at the diff.